qt/scientific_interfaces/Indirect/AbsorptionCorrections.cpp

```diff
@@ -486,7 +490,8 @@ void AbsorptionCorrections::convertSpectrumAxes(
 void AbsorptionCorrections::convertSpectrumAxes(
     const MatrixWorkspace_sptr &correction,
     const MatrixWorkspace_sptr &sample) {
-  if (correction && sample && getEMode(sample) == "Indirect") {
+  if (correction && sample &&
+      sample->getEMode() == DeltaEMode::Type::Indirect) {
```

Comment:
Just to double check is this check supposed to be just Indirect or should it include Direct as well?

Reply:
Well I made the change mainly because the IndirectTab::getEMode function was incorrectly identifying elastic workspaces as Indirect. So I've swapped over to using ExperimentInfo::getEMode() instead. This means the code no longer throws an error trying to convert the spectrum axis on elastic workspaces.

I think the fact that this UI converts the spectrum axis into ElasticQ as a final step for all Indirect workspaces is a bit unnecessary\aggressive so I'm tempted to leave it as it is rather than apply this final step to Direct also. Without this the units on the spectrum axis in the absorption correction result would match the units on the original (uncorrected) workspace - which may well be ElasticQ but could be something else.

Reply:
Well I made the change mainly because the IndirectTab::getEMode function was incorrectly identifying elastic workspaces as Indirect. So I've swapped over to using ExperimentInfo::getEMode() instead. This means the code no longer throws an error trying to convert the spectrum axis on elastic workspaces.

I think the fact that this UI converts the spectrum axis into ElasticQ as a final step for all Indirect workspaces is a bit unnecessary\aggressive so I'm tempted to leave it as it is rather than apply this final step to Direct also. Without this the units on the spectrum axis in the absorption correction result would match the units on the original (uncorrected) workspace - which may well be ElasticQ but could be something else.